packages/cxl-ui/src/components/cxl-marketing-nav.js

```diff
@@ -273,7 +275,29 @@ export class CXLMarketingNavElement extends LitElement {
         );
 
         // Populate.
-        contextMenu.items = this._createContextMenuItems(menuItem.children);
+        const contextMenuItems = this._createContextMenuItems(menuItem.children);
+        contextMenu.items = contextMenuItems;
+
+        if (menuItem.description) {
+          contextMenu.addEventListener('opened-changed', () => {
+            if (document.querySelector('vaadin-context-menu-list-box')) {
+              this._setContextMenuItemDescription(menuItem);
+            }
+          });
+
+          const observe = (mediaQueryString, callback) => {
+            const observer = window.matchMedia(mediaQueryString);
+            const matches = (mediaQueryList) => callback(mediaQueryList.matches);
+            matches(observer);
+            observer.addEventListener('change', matches);
+          };
+
+          observe('(min-width: 420px) and (min-height: 420px)', () => {
```

Comment:
Curious, why do we have to have a seperate description operation for mobile at all - does the context menu get rebuilt? I think I've never had to react to anything on media change for this component, so not sure why it's useful.

Reply:
`vaadin-context-menu` relies on a media change of `420px`

Reply:
Yes, but why wouldn't just populating description once on `opened-changed` event work universally?

Reply:
User on mobile vertical decides to switch to horizontal. Continuity of product.

Reply:
Ahhhh, right, it's about the width update, not content - brain must've been fried, not sure how I missed that.

Reply:
My brain was a little fried too, otherwise I would have been clearer in my explanation :)